File: BotsSDK/korebot/src/main/java/kore/botssdk/fragment/header/ChatHeaderTwoFragment.java
package kore.botssdk.fragment.header;

import static kore.botssdk.view.viewUtils.DimensionUtil.dp1;

import android.content.res.ColorStateList;
import android.graphics.Color;
import android.os.Bundle;
import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.core.content.res.ResourcesCompat;

import com.bumptech.glide.Glide;
import com.bumptech.glide.load.resource.bitmap.CircleCrop;
import com.bumptech.glide.request.RequestOptions;

import kore.botssdk.R;
import kore.botssdk.models.BotResponse;
import kore.botssdk.models.BrandingHeaderButtonsModel;
import kore.botssdk.models.BrandingHeaderModel;
import kore.botssdk.models.BrandingQuickStartButtonActionModel;
import kore.botssdk.net.SDKConfiguration;
import kore.botssdk.utils.BundleConstants;
import kore.botssdk.utils.BundleUtils;

public class ChatHeaderTwoFragment extends BaseHeaderFragment {
    private BrandingHeaderModel brandingModel;

    @Override
    public void setBrandingDetails(BrandingHeaderModel brandingModel) {
        this.brandingModel = brandingModel;
    }

    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.bot_header_2, null);
        TextView tvBotTitle = view.findViewById(R.id.tvBotTitle);
        TextView tvBotDesc = view.findViewById(R.id.tvBotDesc);
        ImageView ivBotAvatar = view.findViewById(R.id.ivBotAvatar);
        LinearLayout llBotAvatar = view.findViewById(R.id.llBotAvatar);
        ImageView ivBotHelp = view.findViewById(R.id.ivBotHelp);
        ImageView ivBotSupport = view.findViewById(R.id.ivBotSupport);
        ImageView ivBotClose = view.findViewById(R.id.ivBotClose);
        ImageView ivBotArrowBack = view.findViewById(R.id.ivBotArrowBack);

        if (brandingModel != null) {
            String title = brandingModel.getTitle() != null ? brandingModel.getTitle().getName() : null;
            tvBotTitle.setText(!TextUtils.isEmpty(title) ? title : SDKConfiguration.Client.bot_name);
            tvBotDesc.setText(brandingModel.getSub_title() != null ? brandingModel.getSub_title().getName() : null);
            view.setBackgroundColor(Color.parseColor(brandingModel.getBg_color()));

            if (brandingModel.getIcon() != null) {
                if (BundleUtils.CUSTOM.equals(brandingModel.getIcon().getType())) {
                    llBotAvatar.setBackgroundResource(0);
                    Glide.with(requireActivity())
                            .load(brandingModel.getIcon().getIcon_url())
                            .apply(RequestOptions.bitmapTransform(new CircleCrop()))
                            .into(ivBotAvatar)
                            .onLoadFailed(ResourcesCompat.getDrawable(getResources(), R.mipmap.ic_launcher, requireContext().getTheme()));

                    LinearLayout.LayoutParams layoutParams = new LinearLayout.LayoutParams((int) (40 * dp1), (int) (40 * dp1));
                    ivBotAvatar.setLayoutParams(layoutParams);
                } else {
                    llBotAvatar.setBackgroundTintList(ColorStateList.valueOf(Color.parseColor(brandingModel.getBg_color())));

                    switch (brandingModel.getIcon().getIcon_url()) {
                        case BotResponse.ICON_1:
                            ivBotAvatar.setImageDrawable(ResourcesCompat.getDrawable(getResources(), R.drawable.ic_icon_1, requireContext().getTheme()));
                            break;
                        case BotResponse.ICON_2:
                            ivBotAvatar.setImageDrawable(ResourcesCompat.getDrawable(getResources(), R.drawable.ic_icon_2, requireContext().getTheme()));
                            break;
                        case BotResponse.ICON_3:
                            ivBotAvatar.setImageDrawable(ResourcesCompat.getDrawable(getResources(), R.drawable.ic_icon_3, requireContext().getTheme()));
                            break;
                        case BotResponse.ICON_4:
                            ivBotAvatar.setImageDrawable(ResourcesCompat.getDrawable(getResources(), R.drawable.ic_icon_4, requireContext().getTheme()));
                            break;
                    }
                }
            }

            if (brandingModel.getButtons() != null) {
                BrandingHeaderButtonsModel buttons = brandingModel.getButtons();
                ColorStateList bgColorTint = ColorStateList.valueOf(Color.parseColor(brandingModel.getIcons_color()));
                ivBotArrowBack.setBackgroundTintList(bgColorTint);

                ivBotHelp.setVisibility(buttons.getHelp() != null && buttons.getHelp().isShow() ? View.VISIBLE : View.GONE);
                ivBotSupport.setVisibility(buttons.getLive_agent() != null && buttons.getLive_agent().isShow() ? View.VISIBLE : View.GONE);
                ivBotClose.setVisibility(buttons.getClose() != null && buttons.getClose().isShow() ? View.VISIBLE : View.GONE);

                ivBotHelp.setBackgroundTintList(bgColorTint);
                ivBotSupport.setBackgroundTintList(bgColorTint);
                ivBotClose.setBackgroundTintList(bgColorTint);

                if (buttons.getHelp() != null) {
                    ivBotHelp.setOnClickListener(v -> onClick(buttons.getHelp().getAction()));
                }
                if (buttons.getLive_agent() != null) {
                    ivBotSupport.setOnClickListener(v -> onClick(buttons.getLive_agent().getAction()));
                }
                if (buttons.getClose() != null) {
                    ivBotClose.setOnClickListener(v -> onClick(buttons.getClose().getAction()));
                }
            }
        }

        ivBotArrowBack.setOnClickListener(v -> requireActivity().getOnBackPressedDispatcher().onBackPressed());
        ivBotClose.setOnClickListener(v -> requireActivity().getOnBackPressedDispatcher().onBackPressed());

        return view;
    }

    private void onClick(BrandingQuickStartButtonActionModel actionModel) {
        if (actionModel != null) {
            switch (actionModel.getType()) {
                case BundleConstants.BUTTON_TYPE_URL:
                case BundleConstants.BUTTON_TYPE_WEB_URL:
                    if (actionModel.getValue() != null) {
                        if (invokeGenericWebViewInterface != null) {
                            invokeGenericWebViewInterface.invokeGenericWebView(actionModel.getValue());
                        }
                    }
                    break;
                default:
                    String actionValue = actionModel.getValue() != null ? actionModel.getValue() : actionModel.getTitle();
                    if (actionValue != null && composeFooterInterface != null) {
                        composeFooterInterface.onSendClick(actionModel.getValue(), false);
                    }
                    break;
            }
        }
    }
}

File: BotSdk/ui/src/main/java/com/kore/ui/botchat/fragment/ChatHeaderTwoFragment.kt
package com.kore.ui.botchat.fragment

import android.content.res.ColorStateList
import android.graphics.Color
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.LinearLayout
import androidx.core.content.res.ResourcesCompat
import androidx.core.view.isVisible
import androidx.fragment.app.Fragment
import com.bumptech.glide.Glide
import com.kore.common.event.UserActionEvent
import com.kore.common.extensions.getParcelableCompat
import com.kore.common.utils.DimensionUtils
import com.kore.event.BotChatEvent
import com.kore.model.constants.BotResponseConstants
import com.kore.network.api.responsemodels.branding.BrandingHeaderModel
import com.kore.network.api.responsemodels.branding.BrandingQuickStartButtonActionModel
import com.kore.ui.R
import com.kore.ui.botchat.BotChatActivity.Companion.EXTRA_BOT_HEADER
import com.kore.ui.databinding.BotHeader2Binding

class ChatHeaderTwoFragment(private val onActionEvent: (event: UserActionEvent) -> Unit) : Fragment() {

    lateinit var binding: BotHeader2Binding

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {

        binding = BotHeader2Binding.inflate(layoutInflater)

        val bundle = this.arguments?.getParcelableCompat(EXTRA_BOT_HEADER) as BrandingHeaderModel?

        if (bundle != null) {
            binding.tvBotTitle.text = bundle.title?.name
            binding.tvBotDesc.text = bundle.subTitle?.name
            binding.root.setBackgroundColor(Color.parseColor(bundle.bgColor))

            if (bundle.icon != null) {
                if (bundle.icon?.type == (BotResponseConstants.CUSTOM)) {
                    binding.llBotAvatar.setBackgroundResource(0)
                    Glide.with(requireActivity()).load(bundle.icon?.iconUrl).into(binding.ivBotAvatar)
                        .onLoadFailed(ResourcesCompat.getDrawable(resources, R.drawable.ic_launcher_foreground, context?.theme))
                    binding.ivBotAvatar.layoutParams =
                        LinearLayout.LayoutParams((40 * DimensionUtils.dp1).toInt(), (40 * DimensionUtils.dp1).toInt())
                } else {
                    when (bundle.icon?.iconUrl) {
                        BotResponseConstants.ICON_1 -> binding.ivBotAvatar.setImageDrawable(
                            ResourcesCompat.getDrawable(resources, R.drawable.ic_icon_1, context?.theme)
                        )

                        BotResponseConstants.ICON_2 -> binding.ivBotAvatar.setImageDrawable(
                            ResourcesCompat.getDrawable(resources, R.drawable.ic_icon_2, context?.theme)
                        )

                        BotResponseConstants.ICON_3 -> binding.ivBotAvatar.setImageDrawable(
                            ResourcesCompat.getDrawable(resources, R.drawable.ic_icon_3, context?.theme)
                        )

                        BotResponseConstants.ICON_4 -> binding.ivBotAvatar.setImageDrawable(
                            ResourcesCompat.getDrawable(resources, R.drawable.ic_icon_4, context?.theme)
                        )
                    }
                }
            }

            bundle.buttons?.let { buttons ->
                val bgColorTint = ColorStateList.valueOf(Color.parseColor(bundle.iconsColor))
                binding.ivBotArrowBack.backgroundTintList = bgColorTint
                binding.ivBotHelp.isVisible = buttons.help?.show == true
                binding.ivBotSupport.isVisible = buttons.liveAgent?.show == true
                binding.ivBotClose.isVisible = buttons.close?.show == true
                binding.ivBotHelp.backgroundTintList = bgColorTint
                binding.ivBotSupport.backgroundTintList = bgColorTint
                binding.ivBotClose.backgroundTintList = bgColorTint
                buttons.help?.let { help -> binding.ivBotHelp.setOnClickListener { onClick(help.action) } }
                buttons.liveAgent?.let { liveAgent -> binding.ivBotSupport.setOnClickListener { onClick(liveAgent.action) } }
                buttons.close?.let { close -> binding.ivBotClose.setOnClickListener { onClick(close.action) } }
            }
        }

        binding.ivBotArrowBack.setOnClickListener {
            onActionEvent(BotChatEvent.OnBackPressed)
        }

        binding.ivBotClose.setOnClickListener {
            onActionEvent(BotChatEvent.OnBackPressed)
        }

        return binding.root
    }

    private fun onClick(actionModel: BrandingQuickStartButtonActionModel?) {
        actionModel?.let { action ->
            when (action.type) {
                BotResponseConstants.BUTTON_TYPE_URL, BotResponseConstants.BUTTON_TYPE_WEB_URL -> {
                    action.value?.let { it1 -> BotChatEvent.UrlClick(it1) }?.let { it2 -> onActionEvent(it2) }
                }

                else -> (action.value ?: action.title)?.let { onActionEvent(BotChatEvent.SendMessage(it)) }
            }
        }
    }
}